keps/1224-lending-limit/README.md

```diff
@@ -0,0 +1,138 @@
+
+### User Stories (Optional)
+
+#### Story 1
+
+In order to ensure the full utilization of resources, we generally set `BorrowingLimit` to max, but this will cause a `ClusterQueue` to run out of its all resources, and make the new incoming job slow to preempt. So we want to reserve some resources for a `ClusterQueue`, so that any incoming jobs in the  `ClusterQueue` can get into work immediately.
+
+### Risks and Mitigations
+
+It's possible that other ClusterQueues in the same cohort may not have enough resources to borrow, because of `LendingLimit`. In other word, `LendingLimit` will influence the resource quota other ClusterQueue's can actually get from `BorrowingLimit`.
+
+## Design Details
+
+### Kueue LendingLimit API
+
+Modify ResourceQuota API object:
+
+```go
+type ResourceQuota struct {
+	[...]
+
+	// lendingLimit is the maximum amount of quota for the [flavor, resource]
+	// combination that this ClusterQueue is allowed to lend the unused
+	// quota to other ClusterQueues in the same cohort.
+	// In total, at a given time, Workloads in a ClusterQueue can consume a
+	// quantity of quota equal to lendingLimit, assuming the other
+	// ClusterQueues in the cohort have enough unused quota.
+	// If null, it means that there is no lending limit.
+	// If not null, it must be non-negative.
+	// lendingLimit must be null if spec.cohort is empty.
+	// +optional
+	LendingLimit *resource.Quantity `json:"lendingLimit,omitempty"`
+}
+```
+
+Also `Lended` is added to `ResourceUsage`.
+
+```go
+type ResourceUsage struct {
+	[...]
+
+    // Lended is quantity of quota that is lended to the cohort. In other
+	// words, it's the unused quota that is under the nominalQuota.
+    Lended resource.Quantity `json:"lended,omitempty"`
+}
+```
+
+
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+None.
+
+#### Unit Tests
+
+- `pkg/cache`: `2023-11-15` - `86.5%`
+- `pkg/controller/core/`: `2023-11-15` - `16.5%`
+- `pkg/metrics/`: `2023-11-15` - `45.7%`
+- `pkg/scheduler/`: `2023-11-15` - `80.6%`
+
+#### Integration tests
+
+<!--
+Describe what tests will be added to ensure proper quality of the enhancement.
+
+After the implementation PR is merged, add the names of the tests here.
+-->
+
+- No new workloads can be admitted when the `LendingLimit` greater than `NominalQuota` or less than `0`.
+- No new workloads in the cohort can borrow when the sum of `LendingLimit` in the cohort does not satisfy the requirement.
+
+### Graduation Criteria
+
+## Implementation History
+
+## Drawbacks
+
+## Alternatives
+
+- `GuaranteedQuota` could do the same. But since we already have the `BorrowingLimit`, it seems more reasonable to use `LendingLimit` that sounds better.
```

Comment:
We considered `GuaranteedQuota` as the field name, but `BorrowingLimit` seems more align with the already exist `LendingLimit`,  so we choice the latter.